### PR TITLE
Update hui-area-card.ts: add more device_classes

### PR DESCRIPTION
**Pull request by ildar170975**
## Proposed change

Added a support for more device_classes:

Sensors:
- carbon_dioxide
- carbon_monoxide
- illuminance
- pm1
- pm25
- pm10
- volatile_organic_compounds
- volatile_organic_compounds_parts

Binary sensors:
- cold
- gas
- heat
- smoke

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
I don't think this will all fit on the area card? What happens if the area has all these sensors?

**Comment by ildar170975** [comment]
Then a user has 2 options:
-- hide a particular entity by card-mod
-- do not assign a particular entity to this area.

Probably in a perfect scenario a user should have a possibility to "untick" a particular entity (-ies) to be displayed on a card.

**Comment by bramkragten** [comment]
Adding the binary device classes is ok I think, as those are only alerts.

If we are going to add more sensor device classes, we should add options to card what to show, we cant expect users to use card mod or exclude entities from an area for this.

**Comment by ildar170975** [comment]
> If we are going to add more sensor device classes

Assume a user have a CO2 & PM25 sensor.
So, with default "temperature" & "humidity" it will be 4 values.

On 1920x1080 screen the current Area card has some place for one more sensor:
изображение

On iPhone SE (375x667) portrait - adding more sensors is not desirable:
изображение

On iPhone SE (375x667) landscape - seems that 2 more sensors may be added:
изображение

Anyway, I hope that adding more classes will be good for the 1st step.
Next we'll try to provide options to "add/exclude entities".

**Comment by bramkragten** [comment]
Can you update the PR based on that?

**Comment by ildar170975** [comment]
@bramkragten 
Updated!
Excluded some sensor classes.
Now only CO2 & PM25 are added.

Did not exclude binary_sensor classes (alerts).

**Review by silamon — COMMENTED** [review]
You may need to run prettier for the linting check to pass.

**Comment by karwosts** [comment]
Should the list of possible classes just be wide open via configuration, and let user specify any arbitrary classes he wants? I don't feel like we should need to define icons here, since it already seems to be defined in other places, e.g:

https://github.com/karwosts/frontend/blob/dev/src/common/const.ts#L160
https://github.com/karwosts/frontend/blob/dev/src/common/entity/binary_sensor_icon.ts#L60-L61

Also when the list of sensors gets too long, it doesn't clip but wraps to a second line. Doesn't line break at clean spots though so doesn't look great at the moment.

File: src/panels/lovelace/cards/hui-area-card.ts [hunk 1]
```diff
@@ -68,21 +74,27 @@ const TOGGLE_DOMAINS = ["light", "switch", "fan"];
...
-    humidity: mdiWaterPercent,
+    humidity: mdiWaterPercent
   },
   binary_sensor: {
+    cold: mdiSnowflake,
+    gas: mdiAlertCircle,
+    heat: mdiFire,
     motion: mdiRun,
     moisture: mdiWaterAlert,
+    smoke: mdiSmokeDetectorVariantAlert
```
[the remark this answers is not in the document]
Comment (ildar170975) [reply src/panels/lovelace/cards/hui-area-card.ts]: Unfortunately, these commas seemed to not solve the issue.
Initially these "terminating" commas were present - and anyway that "prettifier" check gave errors.
So, I finally removed them.

**Comment by ildar170975** [comment]
> Should the list of possible classes just be wide open via configuration, and let user specify any arbitrary classes he wants?

A possible way - do not process a `device_class` at all - just show ALL entities which belong to the area.
A card config UI should allow to select/deselect entities for show.
Implementing this solution is rather difficult for me now, so I proposed a simpler way.

> it doesn't clip but wraps to a second line.

This I will try to fix on the next step.
Currently the card allows to show ONE sensor for some class (like "one average temperature instead of three").
So [here](https://github.com/home-assistant/frontend/pull/18738#issuecomment-1825774000) I provided some estimations for "how many sensors may be shown on the card"; seems rather sufficient for currently supported classes (co2, pm25, temperature, humidity) & viewports. Ofc if a user will try to place more cards in a horizontal stack - there will be less space available; for these cases I will try to propose changes.

File: src/panels/lovelace/cards/hui-area-card.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ildar170975) [reply src/panels/lovelace/cards/hui-area-card.ts]: > the list of sensors line was too long, but that's now short enough.

Never believed a length of a list could be a reason to consider a code as "not pretty" )).
Merged these suggestions!

**Comment by karwosts** [comment]
I think this req can probably be closed, I've added fully customizable sensors and binary sensors in another PR. (#19421)

**Comment by ildar170975** [comment]
Agree

**Pull request closed without merging**